penny_university_frontend/src/reducers/paginate.ts

```diff
@@ -23,27 +24,29 @@ const union = (a: Iterable<any>, b: Iterable<any>) => (
 // Requires three types (request, success, and failure),
 // and a function to map the pagination to a key, e.g. (action) => action.filterName.
 const paginate = ({ types, mapActionToKey }:
-  { types: [Array<string>, Array<string>, Array<string>], mapActionToKey:
-    (action: AnyAction) => string | undefined }) => {
-  if (!Array.isArray(types) || types.length !== 3) {
-    throw new Error('Expected types to be an array of three elements.')
-  }
-  if (!types.every((t) => Array.isArray(t))) {
-    throw new Error('Expected types to be strings.')
-  }
+  {
+    types: {
+      requestTypes: Array<string>, successTypes: Array<string>, failureTypes: Array<string>, deleteTypes: Array<string>
+    },
+    mapActionToKey: (action: AnyAction) => string | undefined
+  }) => {
   if (typeof mapActionToKey !== 'function') {
```

Comment:
is this check still necessary? I would figure that Typescript would enforce this and compile error if it couldn't